examples/release_highlights/plot_release_highlights_1_2_0.py

```diff
@@ -76,3 +76,14 @@
 # The estimator can now run on any Array API compliant libraries such as
 # `CuPy <https://docs.cupy.dev/en/stable/overview.html>`__, a GPU-accelerated array
 # library. For details, see the :ref:`User Guide <array_api>`.
+
+# %%
+# Improved efficiency of many estimators
+# --------------------------------------
+# In version 1.1 the efficiency of many estimators relying on the computation of
+# pairwise distances was greatly improved for float64 dense input. In version 1.2,
+# the efficiency of these estimators was further improved for all combinations of
+# float32/float64 and dense/sparse input. It concerns essentially clustering, manifold
```

Comment:
I think some dense-sparse cases are still not optimized, right?

@jjerphan @Vincent-Maladiere

Reply:
Yes indeed, the dense-sparse and sparse-dense cases were not affected.

Reply:
The Dense-CSR and CSR-dense cases have been optimized for all the supported metrics except Euclidean and Squared euclidean for which we need to craft dedicated Cython routines.

Reply:
I added that it was done for all metrics but euclidean.